src/daal4py.cpp

```diff
@@ -132,6 +147,51 @@ static PyObject * _make_nda_from_homogen(daal::data_management::NumericTablePtr
     return NULL;
 }
 
+template<typename T, int NPTYPE>
+static PyObject *_make_npy_from_data(T * data, size_t n) {
+    npy_intp dims[1] = {static_cast<npy_intp>(n)};
+    PyObject* obj = PyArray_SimpleNewFromData(1, dims, NPTYPE, (void*)data);
+    if (!obj) throw std::invalid_argument("conversion to numpy array failed");
+    return obj;
+}
+
+template<typename T, int NPTYPE>
+static PyObject * _make_nda_from_csr(daal::data_management::NumericTablePtr * ptr)
+{
+    daal::data_management::CSRNumericTable * csr_ptr = dynamic_cast<daal::data_management::CSRNumericTable *>(const_cast<daal::data_management::NumericTable *>((*ptr).get()));
+    if(csr_ptr) {
+        T * data_ptr;
+        size_t * colIndices_ptr;
+        size_t * rowOffsets_ptr;
+        csr_ptr->getArrays<T>(&data_ptr, &colIndices_ptr, &rowOffsets_ptr);
+        size_t n = csr_ptr->getDataSize();
+        T * data_copy = (T*)malloc(n * sizeof(T));
```

Comment:
Why doesn't you use  data from `py_data` after allocate ?  Does Numpy have API for allocate data?